command/task_delete.go

```diff
@@ -0,0 +1,106 @@
+package command
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/go-wordwrap"
+)
+
+const cmdTaskDeleteName = "task delete"
+
+// TaskDeleteCommand handles the `task delete` command
+type taskDeleteCommand struct {
+	meta
+
+	flags *flag.FlagSet
+}
+
+func newTaskDeleteCommand(m meta) *taskDeleteCommand {
+	flags := m.defaultFlagSet(cmdTaskDeleteName)
+	return &taskDeleteCommand{
+		meta:  m,
+		flags: flags,
+	}
+}
+
+// Name returns the subcommand
+func (c taskDeleteCommand) Name() string {
+	return cmdTaskDeleteName
+}
+
+// Help returns the command's usage, list of flags, and examples
+func (c *taskDeleteCommand) Help() string {
+	helpText := fmt.Sprintf(`
+Usage: consul-terraform-sync task delete [options] <task name>
+
+  Task Delete is used to delete existing tasks. Will not delete a task
+  if the task is currently running.
```

Comment:
How do you think the delete error due to task running play at scale? I imagine a bad case when someone needs to delete a task because it's flapping terribly but would have to coincidentally catch the gap when it's not actively running. Would we be able to offer a queued delete that would occur right after the current run completes?

Reply:
I'm still thinking through all the details for offering queued deletes but I think we can do it!

I mostly want to make sure that a run can't be queued up while a delete is in progress. The biggest issue with a run after deletion is that it creates an event that does not have an associated driver, which messes up the overall task status API. The secondary/less important problem is when we add cleanup support, the deleted resources would potentially get recreated. Again, I think we can do it, my brain is just still in catch up mode so I've got to think it through some more.

A workaround we could offer currently is to have the user disable the task, which will wait for the task to complete, and then follow up with a delete. Not an ideal workflow but just throwing it out there in case the queued deletes takes longer to implement than I anticipate!

Reply:
Those are great cases to consider for marking deletion. Thanks for sharing your thoughts there and it's not blocking for this PR since it's CLI focused!